bson/bsonoptions/empty_interface_codec_options.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package bsonoptions
+
+import (
+	"reflect"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+var defaultDecodeType = reflect.TypeOf(primitive.D{})
+
+// EmptyInterfaceCodecOptions represents all possible options for interface{} encoding and decoding.
+type EmptyInterfaceCodecOptions struct {
+	DecodeDefaultType  *reflect.Type // Specifies if the default type for decoding. Defaults to bson.D.
+	DecodeUnpackBinary *bool         // Specifies if Old and Generic type binarys should default to []slice instead of primitive.Binary. Defaults to false.
```

Comment:
New name suggestion: `DecodeBinaryAsByteSlice`? Or if that's too long, `DecodeBinaryAsSlice`?